core/src/main/java/org/mule/runtime/core/event/mutator/AbstractAddVariablePropertyProcessor.java

```diff
@@ -99,4 +104,11 @@ public void setValue(String value) {
     this.valueEvaluator = new AttributeEvaluator(value);
   }
 
+  public void setReturnDataType(DataType type) {
```

Comment:
Are these needed, i don't think they make sense here.  I assume this is because XML element extends abstract-transformer, instead can we change that?